### PR TITLE
Update PointConditionalDistribution

### PR DESCRIPTION
**Pull request by jschueller**
Split from #2795

### CODE REVIEW

File: lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx [hunk 1]
```diff
@@ -222,19 +245,80 @@ class PointConditionalDistributionVBoundEvaluation : public EvaluationImplementa
...
+    if (conditioningIndicesBlocks[i].getSize() == distributions[i].getDimension())
+      continue;
+    // If the block is not conditioned
+    if (conditioningIndicesBlocks[i].getSize() == 0)
+    {
+      newBlocks.add(distributions[i]);
+      continue;
+    }
+    const PointConditionalDistribution conditionalBlock(distributions[i], conditioningIndicesBlocks[i], conditioningValuesBlocks[i]);
+    if (conditionalBlock.useSimplifiedVersion_)
```
Comment (regislebrun) [inline lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx:291]: As discussed during the technical meeting, this line and the lines below can be replaced by
```
newBlocks.add(conditionalBlock.getSimplifiedVersion());
```
as the simplified version is the PointConditionalDistribution if no simplification occurred.

**Review by regislebrun — APPROVED** [review]
Only a minor change

File: lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx (hunk 1 above)
Comment (jschueller) [reply lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx]: I missed that

File: python/test/t_PointConditionalDistribution_std.py [hunk 2]
```diff
@@ -88,3 +88,59 @@
 simplified = distribution.getSimplifiedVersion()
 print(simplified)
 assert simplified.getName() == "JointDistribution", "wrong type"
+
+# special case for Mixture
+mixture = ot.Mixture([ot.Normal(2), ot.Normal(2)], [0.3, 0.7])
```
Comment (josephmure) [inline python/test/t_PointConditionalDistribution_std.py:93]: You could maybe test the PDF/CDF of these distributions?

Comment (jschueller) [reply python/test/t_PointConditionalDistribution_std.py:96]: yes

File: lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx [hunk 3]
```diff
@@ -251,6 +335,122 @@ void PointConditionalDistribution::update()
...
+        if (!start.getDimension())
+          throw InternalException(HERE) << "Could not find a feasible starting point to initialize ration of uniforms V inf";
+        problemVI.setBounds(Interval(lb, zero));
+        algo.setProblem(problemVI);
+        algo.setStartingPoint(start);
+        algo.run();
+        infV_[i] = -std::exp(algo.getResult().getOptimalValue()[0]);
+        LOGDEBUG(OSS() << "infV_[" << i << "]=" << infV_[i] << " v*=" << algo.getResult().getOptimalPoint());
+      }
+    }
```
Comment (josephmure) [inline lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx:437]: According to Codecov this whole block is untested: 

image

File: lib/src/Uncertainty/Distribution/openturns/PointConditionalDistribution.hxx [hunk 4]
```diff
@@ -97,6 +97,12 @@ public:
   Scalar computeConditionalQuantile(const Scalar q, const Point & y) const override;
```
Comment (josephmure) [inline lib/src/Uncertainty/Distribution/openturns/PointConditionalDistribution.hxx:97]: According to Codecov the following methids are untested:
- computeSequentialConditionalPDF 
- computeSequentialConditionalCDF 
- computeConditionalQuantile
- computeSequentialConditionalQuantile
- getIsoProbabilisticTransformation
- getInverseIsoProbabilisticTransformation

**Review by josephmure — APPROVED** [review]

**Pull request merged**